SU2_CFD/src/solver_direct_mean_inc.cpp

```diff
@@ -6128,19 +6172,54 @@ void CIncEulerSolver::SetResidual_DualTime(CGeometry *geometry, CSolver **solver
       
       /*--- Store the residual and compute the Jacobian contribution due
        to the dual time source term. ---*/
-      
+      if (!energy) Residual[nDim+1] = 0.0;
       LinSysRes.AddBlock(iPoint, Residual);
       if (implicit) {
-        for (iVar = 1; iVar < nVar; iVar++) {
-          if (config->GetUnsteady_Simulation() == DT_STEPPING_1ST)
-            Jacobian_i[iVar][iVar] = Volume_nP1/TimeStep;
-          if (config->GetUnsteady_Simulation() == DT_STEPPING_2ND)
-            Jacobian_i[iVar][iVar] = (3.0*Volume_nP1)/(2.0*TimeStep);
+        
+        /*--- Calculating the inverse of the preconditioning matrix
+         that multiplies the time derivative during time integration. ---*/
+
+        /*--- For implicit calculations, we multiply the preconditioner
+        by the cell volume over the time step and add to the Jac diagonal. ---*/
+
+        Jacobian_i[0][0] = 1.0/BetaInc2;
```

Comment:
To be safe and avoid duplication, is it possible to call the SetPreconditioner() routine here to get the matrix? That way, if the matrix changes in the future, we won't introduce bugs here.

Reply:
Yes, I've updated to get the preconditioner matrix directly using SetPreconditioner() for the Jacobian contributions of both static and dynamic grid problems